File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = deque()
        for ch in s:
            if st and st[-1] == '(' and ch == ')': st.pop()
            elif st and st[-1] == '[' and ch == ']': st.pop()
            elif st and st[-1] == '{' and ch == '}': st.pop()
            else: st.append(ch)
        return len(st) == 0

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        for(char ch: s.toCharArray()) {
            if(!st.isEmpty() && st.peek() == '(' && ch == ')') st.pop();
            else if(!st.isEmpty() && st.peek() == '[' && ch == ']') st.pop();
            else if(!st.isEmpty() && st.peek() == '{' && ch == '}') st.pop();
            else st.push(ch);
        }
        return st.isEmpty();
    }
}